webapp/src/ts/services/xml-forms.service.ts

```diff
@@ -68,18 +68,22 @@ export class XmlFormsService {
   }
 
   private getByView(internalId) {
+    let errorMsg = '';
     return this
       .init
       .then(docs => docs?.filter(doc => doc.internalId === internalId))
       .then(docs => {
         if (!docs.length) {
-          return Promise.reject(new Error(`No form found for internalId "${internalId}"`));
+          errorMsg = `No form found for internalId "${internalId}"`;
+          return Promise.reject(new Error(errorMsg));
         }
         if (docs.length > 1) {
-          return Promise.reject(new Error(`Multiple forms found for internalId: "${internalId}"`));
+          errorMsg = `Multiple forms found for internalId: "${internalId}"`;
+          return Promise.reject(new Error(errorMsg));
         }
         return docs[0];
-      });
+      })
+      .catch(err => console.error('Error in XMLFormService getByView : ', errorMsg, err));
```

Comment:
I think that using `err.message` will achieve the same effect _and_ it will not require defining the additional variable. 

Also, is there a reason we don't propagate this error any more? 
 
```suggestion
      .catch(err => {
        console.error('Error in XMLFormService getByView : ', err?.message, err)
        throw err;
      });
```

Reply:
yeah we should propagate and using `err.message` made it simpler